tests/by-util/test_ls.rs

```diff
@@ -6311,3 +6311,220 @@ fn test_ls_time_sort_without_long() {
+    at.touch("visible");
+    at.touch(".hidden");
+
+    scene
+        .ucmd()
+        .arg("-f")
+        .succeeds()
+        .stdout_contains("visible")
+        .stdout_contains(".hidden");
+}
+
+#[test]
+fn test_f_flag_disables_sorting() {
+    // Test that -f disables sorting (equivalent to -U)
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("zebra");
+    at.touch("apple");
+    at.touch("banana");
+
+    // Without -f, files should be sorted alphabetically
+    let sorted = scene.ucmd().succeeds().stdout_move_str();
+    let sorted_lines: Vec<&str> = sorted.lines().collect();
+    assert!(sorted_lines[0].contains("apple"));
+
+    // With -f, files should NOT be sorted
+    let unsorted = scene.ucmd().arg("-f").succeeds().stdout_move_str();
+    assert!(unsorted.contains("zebra"));
+    assert!(unsorted.contains("apple"));
+    assert!(unsorted.contains("banana"));
+}
+
+#[test]
+fn test_f_flag_disables_implicit_color() {
+    // Test that -f disables implicit color (not explicitly set)
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.mkdir("dir");
+    at.touch("file.txt");
+
+    // -f without explicit --color: should not have color
+    let result = scene.ucmd().arg("-f").succeeds().stdout_move_str();
+    assert!(
+        !result.contains("\x1b["),
+        "Color should be disabled with -f alone"
+    );
+}
+
+#[test]
+fn test_explicit_color_always_works_with_f() {
+    // Test that explicit --color always enables color, regardless of -f
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.mkdir("dir");
+    at.touch("file.txt");
+
+    // --color then -f: explicit --color should still enable color
+    let result1 = scene
+        .ucmd()
+        .arg("--color=always")
+        .arg("-f")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        result1.contains("\x1b["),
+        "Explicit --color should work even with -f after"
+    );
+
+    // -f then --color: explicit --color should enable color
+    let result2 = scene
+        .ucmd()
+        .arg("-f")
+        .arg("--color=always")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        result2.contains("\x1b["),
+        "Explicit --color should work even with -f before"
+    );
+}
+
+#[test]
+fn test_f_overrides_a_and_big_a() {
+    // Test last-flag-wins: -f after -a/-A
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("visible");
+    at.touch(".hidden");
+
+    // -A then -f: should show all files
+    scene
+        .ucmd()
+        .arg("-A")
+        .arg("-f")
+        .succeeds()
+        .stdout_contains(".hidden");
+
+    // -f then -A: should honor -A (almost all)
+    let output = scene
+        .ucmd()
+        .arg("-f")
+        .arg("-A")
+        .succeeds()
+        .stdout_move_str();
+    assert!(output.contains(".hidden"));
```

Comment:
Checking for the hidden file is not that useful as it is shown in both cases ;-) Maybe check for `..`? `..` should be shown if `-f` "wins", but not if `-A` wins.